rama-tls/src/boring/client/connector.rs

```diff
@@ -442,15 +442,15 @@ impl<S, K> TlsConnector<S, K> {
                 let store_server_cert_chain = connector_data
                     .is_some_and(|data| data.connect_config_input.store_server_certificate_chain);
 
-                let server_certificate_chain = store_server_cert_chain
-                    .then(|| {
-                        stream
-                            .ssl()
-                            .peer_cert_chain()
-                            .map(|chain| chain.try_into().ok())
-                    })
-                    .flatten()
-                    .flatten();
+                let server_certificate_chain = if store_server_cert_chain {
```

Comment:
I have nothing against flatten btw, just the double flatten and the removal of the error was a bit unwanted, e.g. flatten can play a role here, just in a different way: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2c71e9399ed9a1fc851a18cbb044be54

Reply:
True agree that the initial approach was way too complex. Just lots of different ways to do this and playing around a bit to find the optimal approach, feedback here is greatly appreciated